core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1239,10 +1242,30 @@ private void unParseRegexpLike(SqlWriter writer, SqlCall call, int leftPrec, int
     writer.endFunCall(ifFrame);
   }
 
-  private void unparseShiftLeft(SqlWriter writer, SqlCall call) {
+  private void unparseShiftLeftAndShiftRight(SqlWriter writer, SqlCall call, boolean isShiftLeft) {
     call.operand(0).unparse(writer, 0, 0);
-    writer.print(SHIFTLEFT + " ");
-    call.operand(1).unparse(writer, 0, 0);
+    SqlNode firstOperand = call.operand(1);
+    if (firstOperand instanceof SqlBasicCall
+        && (((SqlBasicCall) firstOperand).getOperator()).kind == SqlKind.MINUS_PREFIX) {
+      // If the second operand is negative, fetch the postive value and change the operator
```

Comment:
check "positive" spelling in comment and also if we create new methods as suggested then these comments can be put on the top of new methods.

Reply:
Done